[website] React Engineer - xCharts role (#38976)
mui/material-ui: docs/src/pages/careers/react-engineer-x-charts.md

We released [MUI X](https://mui.com/x/), a collection of advanced components; [MUI Design kits](https://mui.com/design-kits/), the MUI components available for Figma, Sketch, etc.; and also host [Templates](https://mui.com/templates/), a set of pre-built UI kits.
We've seen rapid growth with all of them (2-3x per year) and we have more to come.
We are building [a low-code platform](https://mui.com/toolpad/) for developers to significantly increase the speed and lower the barrier for creating simple CRUD/dashboard UIs with MUI's libraries.

We are a fully distributed team, spread across multiple time zones.
We work mainly asynchronously, relying mostly on written collaboration.
Every contributor has the freedom to define how they work — the rest of us won't know what you're doing day-to-day unless you tell us.
For additional details about the MUI team and culture, you can check our [careers](https://mui.com/careers/) and [about](https://mui.com/about/) pages and also our [public Handbook](https://mui-org.notion.site/Handbook-f086d47e10794d5e839aef9dc67f324b).

## Why we're hiring

The charts team (part of MUI X) needs your help.

> The component is off to [a great start](https://npm-stat.com/charts.html?package=%40mui%2Fx-charts&from=2021-06-01&to=2023-09-13), however we have only started to scratch the surface for the potential of this component. There is x10 more to build.

Author: Do we want to have a more detailed explanation of what the targets for the charts are?

Author: I've extended the description a bit.

  It would be great if you could address this in your cover letter!

### What it would be nice if you had, but isn't required

- **You've contributed to a design system before**.
  You have worked on enterprise components in the past, maybe for internal use in your company.
  You have built complex and advanced features while making sure the accessibility was fully functional.
- **You've maintained an active repository before**.
  Maybe you've helped maintain a popular open-source repository, or perhaps you've worked on internal repositories that saw contributions from multiple teams.
  Previous experience with highly active repository workflows is a definite plus for this role.
- **You have contributed code to MUI before**. A history of contributing to MUI would be a definite plus.

## Benefits & Compensation

Competitive compensation depending on the profile and location.
We are ready to pay top market rates for a person that can clearly exceed the role's expectations.
You can find the other perks & benefits on the [careers](https://mui.com/careers/#perks-amp-benefits) page.

## How to apply?

> [Apply now for this position 📮](https://jobs.ashbyhq.com/MUI/77584e4c-9379-45f3-a294-6e50bf250383/application?utm_source=ZNRrPGBkqO)

Author: Should we have a dedicated role for `React Engineer - xCharts` on Ashby or we can get away with the current `React Engineer - X` one? My main argument is that if we find someone `who` is a solid candidate but does not have a prior experience with charts we can rotate people around and still be ok with the hire. I'm open to suggestions.

Reviewer: I think the job in Ashby should be a new "React Engineer - xCharts" to match with this page. We can transfer candidates between Ashby job.

> My main argument 

Which option are you arguing for 😁?

Author: My point was if we should have an opening specific for charts on the general MUI X React Engineer one but already discussed it and the charts it is. We can open the React Engineer - X general role just to differentiate the candidates.

We are a fully distributed team, spread across multiple time zones.
We work mainly asynchronously, relying mostly on written collaboration.
Every contributor has the freedom to define how they work — the rest of us won't know what you're doing day-to-day unless you tell us.
For additional details about the MUI team and culture, you can check our [careers](https://mui.com/careers/) and [about](https://mui.com/about/) pages and also our [public Handbook](https://mui-org.notion.site/Handbook-f086d47e10794d5e839aef9dc67f324b).

## Why we're hiring

The charts team (part of MUI X) needs your help.
The component is off to [a great start](https://npm-stat.com/charts.html?package=%40mui%2Fx-charts&from=2021-06-01&to=2023-09-13), however we have only started to scratch the surface for the potential of this component. There is x10 more to build.

To be more specific in what we are aiming to achieve in the short to mid-term:

- A tighter integration with the rest of the MUI libraries including Material UI and the Data Grid
- Faster iteration cycles compared to popular open-source alternatives such as chart.js and Recharts

> Looking a the long term we can likely reach 30% of the downloads of alternative React libraries, considering benchmark data. Developers value having a single touch point, but also one they trust, and one that integrates well in the rest of their app.

Reviewer: 30% is with Material UI only, I'm confident we can do much more.

```suggestion
This should allow us to reach 30% of the downloads of alternative React libraries, considering benchmark data in the short term. Developers value having a single touch point, but also one they trust, and one that integrates well into the rest of their app

In the mid-term, we aspired for this charting liberty to become a standard in the React ecosystem, outgrowing Material UI or any other UI library. We believe we can achieve this with an open-core model. We will match the features available in the alternative open-source projects, licensed as MIT in MUI X Charts. We will then provide extra paid features so developers can use a single charting library, without the need to go to paid alternatives as they do today.

In the long term, we envision using the pro-code charting API as a distribution channel for building more abstract charting primitives, using low-code and AI.
```

> We also need help to continue to make the components easier to use, make it more customizable, [improve performance](https://www.causal.app/blog/react-perf), make it more accessible, improve the health of the open-source by engaging and collaborating with the community, guide developers to answers, and just generally being a positive presence in the community.

Reviewer 2: Are you sure we need this link to causal?

Author: I checked and I wasn't able to find such a blog post from us. We can remove it, it was @oliviertassinari suggestion.

Reviewer: I think we can remove it. It makes sense for me for a data grid, but it feels a bit strange for charts.

- **A track record of demonstrating an eye for product and solving real-world user problems**. If you have a knack for solving problems at the root cause, shipping beautiful user interfaces and intuitive APIs, we want you on our team.
- **Experience building and shipping production code in a team setting** with a passion for writing tested, performant, and high-quality code.
- **Strong written and verbal communication skills**.
  As part of the team, you'll interface both directly and indirectly with community members and enterprise customers, and contribute to user documentation. Clear communication is fundamental in creating intuitive and compelling resources.
- **Ability to dive into complex problems**.
  You should be able to quickly assess, understand, and iterate upon aspects of our codebase.
- **Ready and willing to ask and answer questions**.
  If you're comfortable saying you're unsure, asking for help; but equally reaching out to assist others, you'll be an incredible addition to our team. We thrive because of continuous learning. First-time mistakes should be celebrated, not blamed.

> - **Avoid monolithic deliverables**.

Reviewer 2: ```suggestion
- **Experience or interest in drawing technologies**.
  Charts use SVG and canvas which implies using other API than HTML/CSS. Hope you know them or enjoy discovering new documentation.
- **Avoid monolithic deliverables**.
```

  You scope and stage your work into well-defined milestones to ship.
- **Past work with front-end infrastructure**.
  Perhaps you've created your company's design system, written a Babel plugin, or written complex React components.
  It would be great if you could address this in your cover letter!

### What it would be nice if you had, but isn't required

- **You've contributed to a design system before**.
  You have worked on enterprise components in the past, maybe for internal use in your company.

>   You have built complex and advanced features while making sure the accessibility was fully functional.
> - **You've maintained an active repository before**.

Reviewer 2: ```suggestion
- **You've experience with charts**.
    You have contributed to building chart libraries, integrated charts in a design system, or contributed to chart intensive applications.
    You have built complex and advanced features while making sure the accessibility was fully functional.
- **You've maintained an active repository before**.
```
